src/components/FileExplorerTable/tableFileView.js

```diff
@@ -0,0 +1,154 @@
+import { useState, useEffect } from "react";
+import { useTable, useFilters, usePagination } from "react-table";
+
+
+export const TableFileView = ({ columns, data, stateChanger }) => {
+    // Setting up the State
+    const [filterInput, setFilterInput] = useState("");
+
+    // Set up react-table 
+    const {
+        getTableProps,
+        getTableBodyProps, // Table body props from react-table
+        headerGroups, // Group the table headers
+        prepareRow, // Preparing table row - this function is being called for each row before getting the row props
+        setFilter,
+        // Setting pagination
+        page,
+        canPreviousPage,
+        canNextPage,
+        pageOptions,
+        pageCount,
+        gotoPage,
+        nextPage,
+        previousPage,
+        setPageSize,
+        state: { pageIndex, pageSize },
+    } = useTable({
+        columns,
+        data,
+        initialState: { pageIndex: 0 },
+    },
+        useFilters,
+        usePagination);
+
+
+    // Update the state on changing the search text
+    const handleFilterChange = e => {
+        const value = e.target.value || undefined;
+        setFilter("path", value);
+        setFilterInput(value);
+    };
+
+    //Render Table component with Pagination
+    return (
+        <>
+            <input
+                className="search-input"
+                value={filterInput}
+                onChange={handleFilterChange}
+                placeholder={"Search by file name"}
+            />
+            <table {...getTableProps()}>
```

Comment:
Can the table be styled to resemble the other tables we already have? For instance, take a look at the canary table (components/canary/table.js) at `health` page or the `ConfigListTable` component in `config-list` page